src/mina-signer/src/rosetta.ts

```diff
@@ -49,12 +85,93 @@ function fieldFromHex<T extends Field | Scalar>(
   return [binable.fromBytes(bytes), paddingBit];
 }
 
+function signTransaction(transaction: UnsignedTransaction, privateKey: string, network: NetworkId): SignedRosetta<UnsignedTransaction> {
+  let signature: SignatureJson;
+  if (transaction.payment !== null) {
+    let payment = unsignedTransactionPaymentToPaymentJson(transaction.payment);
+    signature = signPayment(payment, privateKey, network);
+  }
+  else if (transaction.stakeDelegation !== null) {
+    let delegation = unsignedTransactionStakeDelegationToDelegationJson(transaction.stakeDelegation);
+    signature = signStakeDelegation(delegation, privateKey, network);
+  }
+  else {
+    throw Error('signTransaction: Unsupported transaction');
+  }
+  let publicKey = PublicKey.toBase58(PrivateKey.toPublicKey(PrivateKey.fromBase58(privateKey)));
+  return {
+    data: transaction,
+    signature: signatureJsonToHex(signature),
+    publicKey,
+  };
+}
+
+function unsignedTransactionPaymentToPaymentJson(payment: Payment): PaymentJson {
+  return {
+    common: {
+      fee: payment.fee,
+      feePayer: payment.from,
+      nonce: payment.nonce,
+      validUntil: payment.valid_until ?? defaultValidUntil,
+      memo: payment.memo ?? '',
+    },
+    body: {
+      receiver: payment.to,
+      amount: payment.amount,
+    }
+  }
+}
+
+function unsignedTransactionStakeDelegationToDelegationJson(delegation: StakeDelegation): DelegationJson {
+  return {
+    common: {
+      feePayer: delegation.delegator,
+      fee: delegation.fee,
+      validUntil: delegation.valid_until ?? defaultValidUntil,
+      memo: delegation.memo ?? '',
+      nonce: delegation.nonce,
+    },
+    body: {
+      newDelegate: delegation.new_delegate,
+    }
+  }
+}
+
+function verifyTransaction(signedTransaction: SignedRosetta<UnsignedTransaction>, network: NetworkId): boolean {
+  if (signedTransaction.data.payment !== null) {
+    return verifyPayment(unsignedTransactionPaymentToPaymentJson(signedTransaction.data.payment), signatureJsonFromHex(signedTransaction.signature), signedTransaction.publicKey, network);
+  } else if (signedTransaction.data.stakeDelegation !== null) {
+    return verifyStakeDelegation(unsignedTransactionStakeDelegationToDelegationJson(signedTransaction.data.stakeDelegation), signatureJsonFromHex(signedTransaction.signature), signedTransaction.publicKey, network);
+  }
```

Comment:
nit: since you do an early return you can flatten the `else if` to another `if` (would be consistent with the final error not being in an `else` block

Reply:
also, this needs `prettier` :D